### ISSUE TITLE
Multi GPU training and resuming from snapshots is not supported

### ISSUE THREAD

**Issue by Maxfashko** [fizyr/keras-retinanet#944]
Hi guys! How can I get around this limitation?
I have pre-trained weights, and I would like to train again on new data using multi GPUs. Thank you for any help, even if the solution does not look elegant :)

I will also be grateful if you tell me why this restriction is introduced

**Comment by hgaiser** [comment]
Multi-GPU in Keras is a bit weird. Basically what it does is it splits your inputs, copies the network over multiple GPUs and merges the output again. If you would save the `training_model`, you would save this 'splitted and merged' model. This can't be used to train on other hardware configurations, because they might not have the same split/merge requirement.

What you can do though, is save the weights. The only difference is that snapshots also contain their optimizer state (the learning rate, momentum, etc.). If you load weights, you start with a fresh optimizer.
